app/src/main/java/de/westnordost/streetcomplete/map/QuestsMapFragment.kt

```diff
@@ -184,8 +188,9 @@ class QuestsMapFragment : LocationAwareMapFragment() {
             else -> listOf(listOf(g.center))
         }.flatten().all {
             val isContained = controller.latLonToScreenPosition(it, p, false)
-            isContained && p.x >= offset.left && p.x <= mapView.width - offset.right
-              && p.y >= offset.top  && p.y <= mapView.height - offset.bottom
+            val (left, top, right, bottom) = offset
+            val (x, y) = p
+            isContained && x >= left && x <= mapView.width - right && y >= top && y <= mapView.height - bottom
```

Comment:
I think this is not an improvement. It's neither shorter nor easier to read.

Also, it introduces a potential for a bug: Who tells you that the first component is `left` etc? -> Only the documentation. So, this important detail is hidden away from the code, if this code wasn't correct, I wouldn't be able to see it in this review. This `component` stuff in kotlin is not very well thought through in my opinion, it's like using arrays rather than objects to store data, always a potential for bugs.

Reply:
I'll remove that particular commit, then.